### ISSUE TITLE
Technical Implementation for website re-design.

### ISSUE THREAD

**Issue by tgeorgeux** [jupyter/jupyter.github.io#347]
Here's a cross-link to some discussion on the technical side of things with the discourse post: https://discourse.jupyter.org/t/technology-choices-for-new-jupyter-org-website/1672/7

The re-design teams original intent was to build out low fidelity mockups of the new site, find a Premium Gatsby or Hugo template that we could incorporate it into, and then build the Style Guide and Design System around that template around it. It seems like in this case licensing might be an issue. Right now we're looking at whether it makes more sense to try and find/adapt an open-source template or build out a minimal set of components from scratch.

An up-to-date mockup of what we envision the website to be like can be found here: https://www.figma.com/file/PQD2GrejGoeZdYGhXthnAuEs/jupyter_master?node-id=834%3A0 

Part of the reason I'm pushing on a decision here is that the design team has 3 weeks left, and which direction we go will influence what they provide. If we can find a template that 'mostly works' we will massage the design to fit the template if we can't find one and need to build from scratch, we will need to spend the remaining time simplifying the mockups we have instead of adapting them. 

This would also be a good place to discuss if anybody has interest in working on implementing the template or building out the components (depending on which direction we go).

In Summary:
- Please give input/thoughts on the direction we should go.
    - Build out custom website for ourselves.
    - Start with a template we can use based on the current state of mockups (please suggest any you know of). 
- Please let us know if and how you'd like to collaborate on this process. 
- Any other input is welcome.

**Comment by bollwyvl** [comment]
Fantastic work herding the :cat: :joy_cat: :smile_cat: :smirk_cat: !

I'd :heart: to see notebooks represented as an input/control to at least part of the jupyter.org creative pipeline. While a number of individuals/organizations are happily using Nikola, Jekyll, Sphinx, Gatsby and others, with notebooks for the inputs to their personal blogs, textbooks and other "tech heavy" communication channels, there are precious few examples of full implementations of a production "marketing" site.

This of course would bring a number of challenges, such as version control of copy, internationalization, accessibility, etc. but these are the actual problems that our (next X million) users (will) have, and we are in a unique position to demonstrate how to overcome them.

In the most literal sense of accessibility, it doesn't really matter how the actual DOM gets created, but it _is_ important that it scales to screen readers, etc. I'm not too hep to all of the current trend of static site generators, but if it leads to more `npm` and can only be worked on by people that write JSX, i'm pretty :-1:, but that's a personal bias, i suppose.

From an accessibility (broadly-interpreted) point of view, I'd like to steer away from additional proprietary parts of the design process, be it fonts, graphics, cloud collaboration suites, stock photos, clipart, premium templates or plugins: not to belittle the challenges of bringing professional graphic design to open source projects, but if we don't actually own it, we can't fix it, and we can't share it.

**Comment by Carreau** [comment]
Technologically speaking – Don't rebuild from scratch; and –  I would stay with what we have now; and use Jekyll.  The current site use Jekyll so we won't have to spend several hours to get [insert new tech] to work.

Especially in 3 weeks you will likely not get much more done that 1 page well done – which is already a lot; If the first page is done correctly then the other one will just fall in place. 

I would  try to streamline the design to "fundamentals" (for now)  and what is  "nice to have" later. 

For example; and I will go with the home page you have 3 sections:
- 1) Free open-source tools,
- 2) Worldwide community
- 3) Vision for the future. 

- Section (1) "staggers" items; while 2 and 3 don't
- Section (1) has a row of item; sections 2 and 3 don't. 

Can we accept a stepwise design where section (1) does not have these extra design features; and get _that_ in. Refine if there is time.

Another example : remove the "O" as a Jupyter logo in the Hero. It is a extremely costly feature to implement. 

Get the "informations" on the page stripped of all design in a nice datastructure that has as few edge case as possible:

```
section:
  - title: Free and open source tools
  - items:
     -item : JupyterLab
      subtitle: The next Jupyter Notebook
      paragraph: blah blah balh
      link: ./jupyterlab
    - item : JupyterHub
      subtitle: A shared...
      paragraph: blah blah balh
      link: ./jupyterhub
- section:       
  title: Worldwide community.
   ...

```

If this is done, you will realise  inconsistencies (oh ! 90% of the buttons are "Learn More" other have "Donate", "Get involved");

 That will give you:
 - First: the opportunity to simplify (Can we make all the button have the same title for now ?)
 - Second: a base for someone to write the templates. (It's hard to build a template w/o having some data to put in)

With a data structure like above; making a template that renders roughly like figma does  (for home) will be way easier; and it will be simple to improve to reach your final design than to get all at once. 

One one page is 90%, and it works on desktop and mobile; we can start see what is missing for the others. 

> * Please let us know if and how you'd like to collaborate on this process.

I can try to help; but I don't have that much time. As said above for the home page if you give me a rough datasctructure like above I can try to sketch a template.

**Comment by choldgraf** [comment]
Another way to frame what @Carreau suggests - given that we are time-limited on the people with design background, utilizing a new SSG technology now is going to mean a lot of sunk time into figuring out that SSG. Since there's already a Jekyll site running, we can make incremental changes to it much easier.

I'd propose we:

* Table the discussion of "what SSG to use" to a later date
* Focus efforts now on modifying the current template so that it has the design qualities and structure that the design team thinks is good
* Make modifications to content as needed as well
* Document these changes and the new setup

Once we like the structure of the site, it should be much simpler to port to a different SSG (because then the question is just "how do I get this end-product but with a different build system?" instead of solving both questions at the same time)

**Comment by palewire** [comment]
Since this ticket has gone more than two years without any activity, I'm going to close it.
